### PR TITLE
Feature/convention over configuration

### PR DESCRIPTION
**Pull request by felixkempf**
You now only have to config fields which config values differ from the defaults and fields which are not in the table schema (like foreign_keys in other tables, n:m associations,...)

By default, the fields id, created and modified are ignored. If you want to change that, give an array of field names under 'ignoreFields' into the config. See changes in README for examples and explanation. If you want no fields to be ignored, simply pass an empty array under 'ignoreFields'.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -7,13 +7,13 @@
         "installer-name": "ModelHistory"
     },
     "require": {
-        "php": ">=5.4",
+        "php": ">=7.1",
```
Comment (cleptric) [inline composer.json:10]: Just a heads up to release a new major because of that 🙂

File: src/Model/Behavior/HistorizableBehavior.php [hunk 2]
```diff
@@ -78,6 +83,101 @@ public function initialize(array $config)
...
+                            $type = 'association';
+                            break;
+
+                        case $association::ONE_TO_MANY:
+                        case $association::MANY_TO_ONE:
+                            $type = 'relation';
+                            break;
+                    }
+                }
+            } elseif (strpos($columnName, 'password') !== false) {
```
Comment (cleptric) [inline src/Model/Behavior/HistorizableBehavior.php:118]: Should we add a match on stuff that contains `%password%` and `%token%`

File: src/Model/Behavior/HistorizableBehavior.php [hunk 3]
```diff
@@ -78,6 +83,101 @@ public function initialize(array $config)
...
+                'saveable' => true,
+                'obfuscated' => $obfuscated,
+                'type' => $type
+            ], $manualConfig);
+
+            $this->setConfig('fields.' . $columnName, $columnConfig, false);
+        }
+
+        // fill the config array of fields not in the table schema with default values
+        if (!empty($config['fields'])) {
```
Comment (cleptric) [inline src/Model/Behavior/HistorizableBehavior.php:155]: Isn't that a superfluous check?

Comment (felixkempf) [reply src/Model/Behavior/HistorizableBehavior.php:155]: No. If you add the Behavior without any configuration, the key in this array is not even set. That's where the foreach in the line below would complain about undefined index.

Comment (cleptric) [reply src/Model/Behavior/HistorizableBehavior.php:155]:
```
    protected $_defaultConfig = [
        'userIdCallback' => null,
        'entriesToShow' => 10,
        'userNameFields' => [
            'firstname' => 'Users.firstname',
            'lastname' => 'Users.lastname',
            'id' => 'Users.id'
        ],
        'fields' => [], <= ??
        'associations' => [],
        'ignoreFields' => [
            'id',
            'created',
            'modified'
        ]
    ];
```

EDIT: NVM...

File: composer.json (hunk 1 above)
Comment (cleptric) [reply composer.json:10]: Btw, we should add all the typehints then

File: src/Model/Behavior/HistorizableBehavior.php (hunk 2 above)
Comment (felixkempf) [reply src/Model/Behavior/HistorizableBehavior.php:118]: I'm not sure whether "token" is such a dead give away for obscuration like "password" is...

File: composer.json (hunk 1 above)
Comment (felixkempf) [reply composer.json:10]: can do

File: src/Model/Behavior/HistorizableBehavior.php (hunk 2 above)
Comment (cleptric) [reply src/Model/Behavior/HistorizableBehavior.php:118]: `api_token` 😱

File: composer.json (hunk 1 above)
Comment (cleptric) [reply composer.json:10]: https://github.com/scherersoftware/coding-standard -> and use the auto fixer 🙂

File: src/Model/Behavior/HistorizableBehavior.php (hunk 2 above)
Comment (felixkempf) [reply src/Model/Behavior/HistorizableBehavior.php:118]: yeah, but couldn't there be other tokens which are not that sensitive? Any password is always sensitive, but any token?

**Pull request merged**